dashboard/app/controllers/scripts_controller.rb

```diff
@@ -8,7 +8,9 @@ class ScriptsController < ApplicationController
 
   def show
     if @script.redirect_to?
-      redirect_to Script.get_from_cache(@script.redirect_to)
+      redirect_path = script_path(Script.get_from_cache(@script.redirect_to))
+      redirect_query_string = params[:section_id] ? "?section_id=#{params[:section_id]}" : ''
```

Comment:
Would `request.query_string` work here, rather than just checking for this one param?

Reply:
Done, here and in courses controller, although it doesn't clean up the logic as nicely as I would have hoped because it does not include the `?`.

Reply:
also FWIW the script levels controller seems to include any query params already because request.fullpath includes query params.